doc/doxygen/src/kconfig/kconfig.md

```diff
@@ -0,0 +1,366 @@
+
+### 4. Generation of the autoconf.h header
+With the addition of Kconfig a dependency has been added to the build
+process: the `$ (GENERATED_DIR)/autoconf.h` header file. This header file is
+the main output from the Kconfig configuration system. It holds all the macros
+that should be used to configure modules in RIOT:
+`CONFIG_<module>_<parameter>`.
+
+In order to generate the `autoconf.h` file the `genconfig` script is used.
+Inputs for this script are the main `Kconfig` file and `merged.config`
+configuration file, which holds the selected values for the exposed parameters.
+
+#### Input:
+- `$ (GENERATED_DIR)/merged.config` file.
+- Main `Kconfig` file exposing configuration of modules.
+
+#### Output:
+- `$ (GENERATED_DIR)/autoconf.h` configuration header file.
+
+
+### Summary of files
+These files are defined in `kconfig.mk`.
+
+| File              | Description |
+| ----------------- | ----------- |
+| `Kconfig`           | Defines configuration options of modules. |
+| `Kconfig.dep`       | Holds a list of the modules that are being compiled. |
+| `app.config`        | Holds default application configuration values. |
+| `user.config`       | Holds configuration values applied by the user. |
+| `merged.config`     | Holds configuration from multiple sources. Used to generate header. |
+| `autoconf.h`        | Header file containing the macros that applied the selected configuration. |
+
+## Kconfig symbols in Makefiles
+As '.config' files have Makefile syntax they can be included when building.
+This allows to access the applied configuration from the build system. This
+can allow in the future to check for enabled modules.
+
+During migration this is also useful, as it gives the hability to check if a
+parameter is being configured via Kconfig or a default value via `CFLAGS` could
+be injected. For example:
+
+```Makefile
+ifndef CONFIG_USB_VID
+  CFLAGS += -DCONFIG_USB_VID=0x1209
+endif
+```
+
+Symbols will have the same name as the configuration macros (thus will always
+have the `CONFIG_` prefix). As the configuration file is loaded in
+`Makefile.include` care should be taken when performing checks in the
+application's Makefile. The symbols will not be defined until after including
+`Makefile.include`.
+
+---
+# Transition phase                                  {#kconfig-transition-phase}
+## Making configuration via Kconfig optional  {#kconfig-configuration-optional}
+During transition to the usage of Kconfig as the main configurator for RIOT,
+the default behavior will be the traditional one: expose configuration options
+in header files and use CFLAGS as inputs. To allow optional configuration via
+Kconfig, a convention will be used when writing Kconfig files.
+
+Modules should be contained in their own `menuconfig` entries, this way the user
+can choose to enable the configuration via Kconfig for an specific module.
+These entries should depend on the module being used (see
```

Comment:
Maybe say it more explicit: They should define a dependency to one specific module